Assembler - Sidenav - Checklist feature

miksansegundo opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/75156

## Proposed Changes

* Added new props `checklist` and `checked` to the component `<NavigationButtonAsItem>`
* The screens must be completed by choosing an item (header, footer, sections, font, color) to become checked ☑️
* The non-checked styles remain the same

https://github.com/Automattic/wp-calypso/assets/1881481/a1e999aa-b9c4-4cd5-acd8-edaa3de09ee7

**Updated screenshots**
| NO CHECKED| CHECKED | + HOVER | + FOCUS |
|--|--|--|--|
|<img width="331" alt="Screenshot 2566-06-15 at 11 20 27" src="https://github.com/Automattic/wp-calypso/assets/1881481/a237508a-670c-41aa-8d19-94f3de994334">|<img width="339" alt="Screenshot 2566-06-15 at 11 19 19" src="https://github.com/Automattic/wp-calypso/assets/1881481/51b4db9a-4b67-4907-b651-5e02f738f3a0">|<img width="333" alt="Screenshot 2566-06-15 at 11 19 31" src="https://github.com/Automattic/wp-calypso/assets/1881481/80e57d45-58b8-4275-a53b-1216d5477245">|<img width="339" alt="Screenshot 2566-06-15 at 11 20 02" src="https://github.com/Automattic/wp-calypso/assets/1881481/95f23cd6-0924-4f52-b22c-91ff9ab1bc8f">|

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Access the assembler `/setup/with-theme-assembler/patternAssembler?siteSlug={ YOUR SITE }&theme=blank-canvas-3`
* Verify the navigator button icons change to a checkmark when users choose an item from each screen 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [X] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [X] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [X] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [X] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [X] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

miksansegundo committed:
Navigator button checklist feature

miksansegundo commented:
Hi @lucasmendes-design, this is ready for testing!

taipeicoder commented on client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx:
```diff
@@ -135,6 +151,9 @@ const ScreenMain = ( {
 								<span className="pattern-layout__list-item-text">{ translate( 'Colors' ) }</span>
 							</NavigationButtonAsItem>
 							<NavigationButtonAsItem
+								key="font-pairings"
```
Curious why does this button have `key` and the others don't 🤔

taipeicoder reviewed (approved):
LGTM! Some minor feedbacks:

- Could we try setting `pattern-layout__list-item-text` `line-height: 24px`? Seems to vertically align with the check icon better.
- The `:focus` text color for checked items is still blue, should it be `var(--studio-gray-100);`?

 
![Screenshot 2023-06-14 at 10 16 39 AM](https://github.com/Automattic/wp-calypso/assets/797888/63fd6aac-a2b0-427d-9a0b-4ba772a6b006)

taipeicoder commented:
> Hi @lucasmendes-design, this is ready for testing!

We can merge this first and iterate if @lucasmendes-design has further suggestions 🙂

arthur791004 commented on client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/navigator-buttons/index.tsx:
```diff
@@ -44,10 +46,24 @@ const GenericButton = ( { icon, children, ...props }: Props ) => {
 		<NavigatorButton
-			as={ GenericButton }
+			as={ checklist ? ChecklistButton : GenericButton }
 			className={ classnames( 'navigator-button', className ) }
```
We can simply pass the props you want here, and then we don't need the `ChecklistButton`

lucasmendes-design commented:
Thanks folks, I was wrapping up the landing page post. Sorry for the delay. 

Some notes:
- Let's not use the light gray here, I think it's not the right concept, users can go back a play again with header, sections, etc. So, with the light gray, it seems a disabled component. All colors should be black. The hover will be blue. 
<img width="339" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/f4fd83aa-aa32-4d07-aa44-6f16730c284a">
<img width="336" alt="image" src="https://github.com/Automattic/wp-calypso/assets/98343583/36c9fa90-9133-4d30-8b08-01467c46d73b">

miksansegundo committed:
Code and style improvements

miksansegundo commented:
I changed the styles and updated the screenshots. Thanks for the review!

@taipeicoder I don't see any difference when using `line-height: 24px`. Can you review it again? Here is a screenshot with border just to help to see the alignment:

<img width="288" alt="Screenshot 2566-06-15 at 11 02 08" src="https://github.com/Automattic/wp-calypso/assets/1881481/d526d81d-c553-41ec-bd74-20017430f98f">

miksansegundo commented on client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx:
```diff
@@ -100,47 +108,57 @@ const ScreenMain = ( {
 							aria-label={ translate( 'Header' ) }
 							onClick={ () => onSelect( 'header' ) }
 						>
-							<span className="pattern-layout__list-item-text">{ translate( 'Header' ) }</span>
```
The class `.pattern-layout__list-item-text` is only used in the screen-section in the component `PatternLayout` so I'm removing it from the here (the main-screen) and adding a class `.navigator-button__text` in NavigatorButton. cc: @arthur791004

miksansegundo commented on client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/navigator-buttons/style.scss:
```diff
@@ -34,6 +34,28 @@
+			}
+
+			.navigator-button__text {
+				line-height: 24px;
```
This `line-height` is not making any difference. I'll remove it

arthur791004 commented on client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/navigator-buttons/index.tsx:
```diff
@@ -5,28 +5,35 @@ import {
-	children: JSX.Element;
+	children: JSX.Element | string;
 	onClick?: () => void;
+	checklist?: boolean;
```
I guess we don't need this anymore 🙂

arthur791004 reviewed (approved):
Tested and looks good to me!

![image](https://github.com/Automattic/wp-calypso/assets/13596067/43ee0e89-77c6-4585-a687-7f3f147d9342)

taipeicoder commented:
> I changed the styles and updated the screenshots. Thanks for the review!
> 
> @taipeicoder I don't see any difference when using `line-height: 24px`. Can you review it again? Here is a screenshot with border just to help to see the alignment:
> 
> <img alt="Screenshot 2566-06-15 at 11 02 08" width="288" src="https://user-images.githubusercontent.com/1881481/245994037-d526d81d-c553-41ec-bd74-20017430f98f.png">

You're right, I just checked again and it seems aligned just fine 🙂

miksansegundo committed:
Remove type

arthur791004 commented on client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-main.tsx:
```diff
@@ -100,47 +108,57 @@ const ScreenMain = ( {
 				<HStack direction="column" alignment="top" spacing="4">
 					<NavigatorItemGroup title={ translate( 'Layout' ) }>
 						<NavigationButtonAsItem
+							checklist
```
Not just for type 😉

miksansegundo committed:
Remove checklist prop
